lib/azure/core/http/http_request.rb

```diff
@@ -147,10 +140,7 @@ def body=(body)
         # @return [HttpResponse]
         def call
           conn = http_setup
-          res = conn.run_request(method.to_sym, uri, nil, nil) do |req|
-            req.body = body if body
-            req.headers = headers if headers
-          end
+          res = set_up_response(method.to_sym, uri, conn, headers ,body)
 
```

Comment:
@vishrutshah since set_up_response method returns Faraday::Response, I believe `:timeout` is correct based on http://stackoverflow.com/questions/10322283/what-is-timeout-and-open-timeout-in-faraday (set_up_response method was introduced by https://github.com/Azure/azure-sdk-for-ruby/pull/335/files)